### ISSUE TITLE
Passthrough to github?

### ISSUE THREAD

**Issue by mrjoelkemp** [zencephalon/DeftDraft-Sinatra#1]
What do you think about using github as the datastore for drafts? I'm just brainstorming, but for every draft, have the client send it to the backend. The backend then does a background upload to that user's directory in your project's github repo.

With your recent overhaul, I think you did away with git-managed, local directories on the backend. This proposition eliminates the need to have a mongo store of the drafts.

You're now introducing potential network overhead, but you could fetch the user's stuff on the start of the session.

Thoughts?

**Comment by zencephalon** [comment]
Hey Joel!

Yeah, I'd actually considered using Github as a backend using their API. The problem I had with it is just that Github has a bit of latency (although it is one of the fastest services I've ever used). One the other hand, there are a lot of cool features that I'd get for free by doing it. One tricky issue is to do branches I would want to make every document it's own repository. That could be okay if every user signed up for a github, but even then it would be a lot of repos. And starting a repo is relatively high latency.

For the time being I'm just going to store things in Mongo, because text really isn't that expensive to store, and I'll actually be storing diffs for the most part... although not as efficiently as git.

**Comment by mrjoelkemp** [comment]
I'm building something similar and used https://code.google.com/p/google-diff-match-patch/ for diffing. It works very well.

**Comment by zencephalon** [comment]
I'm curious what kind of text you'll be diffing over, because from what I can tell Patience Diff is superior.

**Comment by mrjoelkemp** [comment]
Hmm, I'm diffing over code. From my tests, google's diff works very well,
but I could be proven wrong with a much longer source file.

I'll keep Patience diff in mind. Thanks!

**Comment by zencephalon** [comment]
Yeah, definitely check it out because patience is much better for code than
Meyers.

On Sunday, June 2, 2013, Joel Kemp wrote:

> Hmm, I'm diffing over code. From my tests, google's diff works very well,
> but I could be proven wrong with a much longer source file.
> 
> I'll keep Patience diff in mind. Thanks!
> 
> On Sun, Jun 2, 2013 at 2:11 PM, Matthew Bunday <notifications@github.com<javascript:_e({}, 'cvml', 'notifications@github.com');>>wrote:
> 
> > I'm curious what kind of text you'll be diffing over, because from what
> > I
> > can tell Patience Diff is superior.
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/zencephalon/prosedy/issues/1#issuecomment-18810728>
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/zencephalon/prosedy/issues/1#issuecomment-18811285
> .

## 

Matthew Bunday | Developer | http://zencephalon.com
mbunday@sailthru.com |(952) 201-5074
SAILTHRU http://www.sailthru.com/

http://www.sailthru.com/
